### ISSUE TITLE
Data source oci_containerengine_node_pools does not include node information

### ISSUE THREAD

**Issue by Julppu** [oracle/terraform-provider-oci#1739]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version and Provider Version

``` shell
$ terraform version
Terraform v1.2.2
on darwin_amd64
+ provider registry.terraform.io/oracle/oci v4.102.0
```

### Affected Resource(s)

`oci_containerengine_node_pools`

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    oci = {
      source  = "oracle/oci"
      version = "4.102.0"
    }
  }
}

variable "tenancy_id" {
  type      = string
  sensitive = true
}
variable "user_id" {
  type      = string
  sensitive = true
}
variable "api_key_fingerprint" {
  type      = string
  sensitive = true
}
variable "api_private_key_path" { type = string }
variable "region" { type = string }
variable "compartment_id" { type = string }

provider "oci" {
  tenancy_ocid     = var.tenancy_id
  user_ocid        = var.user_id
  private_key_path = var.api_private_key_path
  fingerprint      = var.api_key_fingerprint
  region           = var.region
}

data "oci_containerengine_node_pools" "pools" {
  compartment_id = var.compartment_id
}

output "pools" {
  value = data.oci_containerengine_node_pools.pools.node_pools.*.nodes
}
```

### Debug Output

<https://gist.github.com/Julppu/25cb33758edf1777fe9eb28587e56348>

### Expected Behavior

Terraform should output details of all nodes belonging to nodepools discovered in the compartment. In this instance there are two node pools and six total nodes.

### Actual Behavior

The output is two empty nested lists, one for each nodepool.

### Steps to Reproduce

Applying the configuration described above.

1. `terraform apply`

### Important Factoids

The desired functionality can (and has to) be done using data source `oci_containerengine_node_pool` instead, retrieving each node pool using pool OCIDs received with `oci_containerengine_node_pools`.

``` hcl
data "oci_containerengine_node_pools" "pools" {
  compartment_id = data.oci_identity_compartment.compartment.id
}

data "oci_containerengine_node_pool" "single_pool" {
  count        = length(data.oci_containerengine_node_pools.pools.node_pools)
  node_pool_id = data.oci_containerengine_node_pools.pools.node_pools[count.index].id
}

output "single_pools" {
  value = data.oci_containerengine_node_pool.single_pool.*.nodes
}
```

**Comment by XinruXiao-9** [comment]
The doc is updated in 4.123.0

**Comment by Maxrovr** [comment]
Fixed in version 4.123.0 by @XinruXiao-9
